source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -0,0 +1,70 @@
+#include "source/common/network/listener_filter_buffer_impl.h"
+
+#include <string.h>
+
+namespace Envoy {
+namespace Network {
+
+uint64_t ListenerFilterBufferImpl::copyOut(Buffer::Instance& buffer, uint64_t length) {
+  auto size = std::min(length, data_size_);
+  if (size == 0) {
+    return 0;
+  }
+  buffer.drain(buffer.length());
```

Comment:
Does emptying the callers buffer make sense?

Reply:
Oh..right, the caller should empty the buffer.